packages/clerk-js/src/core/clerk.ts

```diff
@@ -873,6 +876,23 @@ export default class Clerk implements ClerkInterface {
         buildURL({ base: displayConfig.signUpUrl, hashPath: '/continue' }, { stringify: true }),
     );
 
+    const navigateToNextStepSignUp = () => {
+      if (su.unverifiedFields.length === 0 || su.missingFields.length !== 0) {
+        return navigateToContinueSignUp();
```

Comment:
:question: Is calling `navigateToContinueSignUp()` with no arguments the correct way to proceed here? I would expect that `navigateToNextStepSignUp` is built like the other navigation functions.

Why don't we put this logic inside `SignUpContinue`?

Reply:
> ❓ Is calling `navigateToContinueSignUp()` with no arguments the correct way to proceed here? I would expect that `navigateToNextStepSignUp` is built like the other navigation functions.

Hmm, I'm not sure I get this. This function combines the functionality needed to determine the next step, making it reusable below. The other navigation functions don't have any logic.

> Why don't we put this logic inside `SignUpContinue`?

Do you mean the actual component? The `handleRedirectCallback` seems like a more appropriate place as it determines when to redirect next after the SSO callback.

Reply:
> Why don't we put this logic inside SignUpContinue?

Please ignore this comment, it's a leftover. I had written it initially, but then read the PR more carefully and understood better what we're doing here. Sorry for the confusion.

My question about the `navigateToNextStepSignUp` is about making sure we're not missing any arguments that we might need to pass to navigateToContinueSignUp, such as the signUpUrl. Upon closer inspection, I see that this is already included by the `makeNavigate` call above. So, again, probably a void comment.